src/Microsoft.PowerShell.Commands.Utility/commands/utility/sort-object.cs

```diff
@@ -239,13 +245,12 @@ protected override void EndProcessing()
             // Track the number of items that will be output from the data once it is sorted
             int sortedItemCount = dataToProcess.Count;
 
-            // If -Top & -Bottom were not used, or if -Top or -Bottom would return all objects, invoke
-            // an in-place full sort
-            if ((Top == 0 && Bottom == 0) || Top >= dataToProcess.Count || Bottom >= dataToProcess.Count)
```

Comment:
Why did you remove `Top >= dataToProcess.Count || Bottom >= dataToProcess.Count` condition?

Reply:
The discussion in bug #7846 describes this in detail (and this was the original driver for this PR). With the old logic, when Top or Bottom exceed the number of items being sorted, you wouldn't end up with a heap sort, which you should when you use Top or Bottom. The end result was a less performant, unstable full sort when you expect a stable heap sort to be performed. Removing these conditions fixes that issue (one part of this PR). Adding -Stable addresses the need for stable sorting when you don't care about the original data size (the other part of this PR).